File: #213_House_robber_II/Solution.java
public class Solution {
    public int rob(int[] nums) {
        if (nums.length == 0) return 0;
        if (nums.length == 1) return nums[0];
        int len = nums.length;
        int max = 0;
        for (int i = 0; i < len; i++) {
	        int[] dp = new int[len + 1];
	        int j = i;
	        for (int c = 0; c < len; c++) {
	        	if (c == len - 1) {
	        		dp[c + 1] = dp[c];
	        	}
	        	else if (j == i) dp[c + 1] = nums[j];
	        	else {
	        		int possible = dp[c - 1] + nums[j];
	        		dp[c + 1] = (dp[c] > possible) ? dp[c] : possible;
	        	}
	        	j = (j + 1) % len;
	        }
	        if (dp[dp.length - 1] > max) max = dp[dp.length - 1];
        }
        return max;
    }

    public static void main(String[] args) {
    	int[] test = {1, 1, 1};
    	Solution sol = new Solution();
    	System.out.println(sol.rob(test));
    }
}

File: #213_House_robber_II/Solution.py
class Solution(object):
    def rob(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
        	return 0
        if len(nums) == 1:
        	return nums[0]
        return max(self.subRob(nums[:len(nums) - 1]), self.subRob(nums[1:]))

    def subRob(self, nums):
    	if not nums:
    		return 0
    	rob, skip = 0, 0
    	for val in nums:
    		tmp = rob
    		rob = val + skip
    		skip = max(skip, tmp)
    	return max(rob, skip)

sol = Solution()
print sol.rob([1, 1])